src/main/java/org/rumbledb/runtime/functions/datetime/CurrentDateFunctionIterator.java

```diff
@@ -0,0 +1,39 @@
+package org.rumbledb.runtime.functions.datetime;
+
+import org.joda.time.DateTime;
+import org.rumbledb.api.Item;
+import org.rumbledb.exceptions.ExceptionMetadata;
+import org.rumbledb.exceptions.IteratorFlowException;
+import org.rumbledb.items.DateItem;
+import org.rumbledb.items.ItemFactory;
+import org.rumbledb.runtime.RuntimeIterator;
+import org.rumbledb.runtime.functions.base.LocalFunctionCallIterator;
+
+import sparksoniq.jsoniq.ExecutionMode;
+
+import java.util.List;
+
+public class CurrentDateFunctionIterator extends LocalFunctionCallIterator {
+
+    private static final long serialVersionUID = 1L;
+
+    public CurrentDateFunctionIterator(
+            List<RuntimeIterator> arguments,
+            ExecutionMode executionMode,
+            ExceptionMetadata iteratorMetadata
+    ) {
+        super(arguments, executionMode, iteratorMetadata);
+    }
+
+    @Override
+    public Item next() {
+        if (this.hasNext) {
+            this.hasNext = false;
+            return new DateItem(new DateTime(), true);
```

Comment:
This needs to go through the factory -- this is why constructors are not public.

Like so:

ItemFactory.getInstance().createDateItem(new DateTime(), true);

(the same goes for the other three functions).

Thanks!

Reply:
Thank you for the comment, I made the changes just now.